src/jot.gleam

```diff
@@ -1277,15 +1447,67 @@ fn take_list_item_chars(
   }
 }
 
-fn take_paragraph_chars(in: String) -> #(String, String) {
-  case string.split_once(in, "\n\n") {
+fn take_paragraph_chars(
+  in: String,
+  div_close_size: Option(Int),
+) -> #(String, String) {
+  let #(paragraph, rest) = case string.split_once(in, "\n\n") {
     Ok(#(content, in)) -> #(content, in)
     Error(_) ->
       case string.ends_with(in, "\n") {
         True -> #(string.drop_end(in, 1), "")
         False -> #(in, "")
       }
   }
+
+  case div_close_size {
+    Some(size) -> {
+      let #(split_paragraph, paragraph_rest) =
+        search_paragraph_for_div_end(paragraph, [], size)
+
+      case split_paragraph, paragraph_rest {
+        "", "" -> #(paragraph, rest)
+        _, "" -> #(split_paragraph, rest)
+        _, _ -> #(split_paragraph, paragraph_rest <> "\n\n" <> rest)
+      }
+    }
+    None -> #(paragraph, rest)
+  }
+}
+
+/// search a stretch of paragraph characters for valid paragraph terminator
+///
+/// regex expression: `\n[ \t]*:::+[ \t]*(\n|$)`
```

Comment:
Don't think we need to have regex in the comments.

Capital letter please, and explain what a paragraph terminator is.